app/js/lib/foxbox/foxbox.js

```diff
@@ -389,6 +389,71 @@ export default class Foxbox extends Service {
     return this[p.db].setTag.apply(this[p.db], arguments);
   }
 
+  subscribeToNotifications(resubscribe = false) {
+    const settings = this[p.settings];
+    const boxHost = this[p.net].origin;
+    if (!navigator.serviceWorker) {
+      return Promise.reject('No service worker supported');
+    }
+
+    if (settings.pushEndpoint && settings.pushPubKey && !resubscribe) {
+      return Promise.resolve();
+    }
+
+    return navigator.serviceWorker.ready.then((reg) => {
+      reg.pushManager.subscribe({userVisibleOnly: true})
+      .then((subscription) => {
+        const endpoint = subscription.endpoint;
+        const key = subscription.getKey ? subscription.getKey('p256dh') : '';
+        settings.pushEndpoint = endpoint;
+        settings.pushPubKey = btoa(String.fromCharCode.apply(null,
+          new Uint8Array(key)));
+
+        // Send push information to the server
+        // XXX: We will need some library to write taxonomy messages
+        const pushConfigurationMsg = [[
+            [{
+              id: 'setter:subscribe.webpush@link.mozilla.org'
+            }], {
+              Json: {
+                subscriptions: [{
+                  public_key: settings.pushPubKey,
+                  push_uri: settings.pushEndpoint
+                }]
+              }
+            }
+          ]];
+        return this[p.net].fetchJSON(
+          `${boxHost}/api/v1/channels/set`, 
```

Comment:
nit: We use `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/` everywhere else in this file. If it's too big, at least make sure to use the api version from the settings rather than hardcoding it.